### PR TITLE
Correct result of Stem(), value of $*

### LINKED ISSUE #243: $* gives incorrect value when the stem is in the middle of the pattern
If an output pattern has both a prefix and a suffix, `$*` will contain the suffix as well as the stem.

[example.txt](https://github.com/google/kati/files/8192211/example.txt)

This test outputs:

```
a
b
c
```

in GNU make, but:

```
a.a
b
c
```

in ckati.

### PR DESCRIPTION
**Pull request by DelilahHoare**
Fixes #243

### CODE REVIEW

File: testcase/stem_middle.mk [hunk 1]
```diff
@@ -0,0 +1,11 @@
+
+test: a/a.a b.b c/c
```
Comment (lberki) [inline testcase/stem_middle.mk:2]: Can you make this test fail at HEAD?

**Review by lberki — COMMENTED** [review]
Thanks for the change! It's a nice fix. What do you need this for?

File: testcase/stem_middle.mk (hunk 1 above)
Comment (DelilahHoare) [reply testcase/stem_middle.mk:2]: It does

**Comment by DelilahHoare** [comment]
> What do you need this for?

Linux has a rule `$(obj)/%.h` which invokes the command `echo "\#include <asm-generic/$*.h>" > $@` to generate wrapper files.  With the incorrect result for Stem(), the wrappers include nonexistent files like `#include <asm-generic/types.h.h>` and compilation fails.

File: testcase/stem_middle.mk (hunk 1 above)
Comment (lberki) [reply testcase/stem_middle.mk:2]: just as embarrassing (for me) as on the other thread

**Review by lberki — APPROVED** [review]

**Pull request merged**